Umbraco UIBuilder patches & minor versions update (#5873)
umbraco/UmbracoDocs: 13/umbraco-ui-builder/release-notes.md

This section contains the release notes for Umbraco UI Builder 13 including all changes for this version.

> #### [**13.1.0-rc1**](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F13.1.0) **(February 20th 2024)**

Reviewer: Are the dates right for all the release notes?  Feb 20 was yesterday.

Author: Correct. The packages are on NuGet.

> 

Reviewer: Maybe you could just "editorialise" this a bit more, given it'll be the landing for the social media posts.  Just an introductory sentence or two introducing the focus of the release.  E.g.

"Umbraco UI Builder 13.1 is the first release since launch adding new features.  We've focussed on improving the experience when working with related collections, addressing some additional use cases that widen the scope of the product."

* Retrieve Child Collections

This feature addresses a **one-to-many** relationship context, where having a parent collection with child sub-collections, one might need to retrieve the child collections only, without fetching the details of the parent.

You can read more in [this](collections/retrieve-child-collections.md) section of the documentation.

> * Related Collections

Reviewer: Move this one first I think, as it's probably the biggest addition.

This feature provides support for managing **many-to-many** relationships by configuring main, related and junction entities.

You can read more in [this](collections/related-collections.md) section of the documentation.

> * Implementation of an `UdiConverter`

Reviewer: ```suggestion
* Implementation of a `UdiConverter`
```

When configuring a collection to use as foreign key a reference to an Umbraco entity, while having the foreign key represented as `Integer`, the UDI value of the entity cannot be converted from `String` to `Int`, and will default to 0. With this feature, based on the UDI value, we are retrieving and persisting the `Id` of the Umbraco entity.

> An use case can be found [here](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues/86).

Reviewer: ```suggestion
A use case can be found [here](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues/86).
```

This section contains the release notes for Umbraco UI Builder 13 including all changes for this version.

#### [**13.1.0-rc1**](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F13.1.0) **(February 20th 2024)**

Umbraco UI Builder 13.1 is the first release since launch adding new features. We've focussed on improving the experience when working with related collections, addressing some additional use cases that widen the scope of the product. This includes the possibility to retrieve child collections entities or use an Umbraco entity as foreign key.

* Related Collections

This feature provides support for managing **many-to-many** relationships by configuring main, related and junction entities.

You can read more in [this](collections/related-collections.md) section of the documentation.

* Retrieve Child Collections

This feature addresses a **one-to-many** relationship context. Having a parent collection with child sub-collections, one might need to retrieve the child collections only, without fetching the details of the parent.

> You can read more in [this](collections/retrieve-child-collections.md) section of the documentation.

Reviewer 2: ```suggestion
You can read more about this in the [retrieve child collections](collections/retrieve-child-collections.md) article.
```

* Implementation of a `UdiConverter`

> This update addresses the configuration of collections that use as foreign key a reference to an Umbraco entity. If the FK type is `Integer`, the persisted value defaults to 0. Because the UDI value of the entity cannot be converted from `String` to `Int`. Based on the UDI value, we are retrieving and persisting the `Id` of the Umbraco entity.

Reviewer 2: ```suggestion
This update addresses the configuration of collections that use as foreign key a reference to an Umbraco entity. If the FK type is `Integer`, the persisted value defaults to 0. This is because the UDI value of the entity cannot be converted from `String` to `Int`. Based on the UDI value, we are retrieving and persisting the `Id` of the Umbraco entity.
```

> A use case can be found [here](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues/86).

Reviewer 2: ```suggestion
A use case can be found in the [Github issue #86](https://github.com/umbraco/Umbraco.UIBuilder.Issues/issues/86).
```
